src/utils/installPackage.js

```diff
@@ -72,19 +75,21 @@ async function installPeerDependencies(filepath, name, options) {
 }
 
 async function determinePackageManager(filepath) {
-  let configFile = await config.resolve(filepath, [
-    'yarn.lock',
-    'package-lock.json'
+  const [npmLockFile, yarnLockFile, hasYarn] = await Promise.all([
+    config.resolve(filepath, [PACKAGE_LOCK]),
+    config.resolve(filepath, [YARN_LOCK]),
+    checkForYarnCommand()
```

Comment:
As I mentioned in my previous comment this checkYarn function is a pretty expensive operation and will slow down things. It is unnecessary if there isn't a yarn.lock and therefore would use processing power and slow down parcel for no reason.

resolving configs on the other hand are very fast.